lightkube/core/client.py

```diff
@@ -176,6 +179,76 @@ def watch(self, res, *, namespace=None, labels=None, fields=None, server_timeout
         )
         return self._client.watch(br, on_error=on_error)
 
+    @overload
+    def wait(
+        self,
+        res: Type[GlobalResource],
+        name: str,
+        *,
+        success_conditions: Iterable[str] = DEFAULT_SUCCESS_CONDITIONS,
+        failure_conditions: Iterable[str] = DEFAULT_FAILURE_CONDITIONS,
+    ) -> GlobalResource:
+        ...
+
+
+    @overload
+    def wait(
+        self,
+        res: Type[AllNamespacedResource],
+        name: str,
+        *,
+        namespace: str = None,
+        success_conditions: Iterable[str] = DEFAULT_SUCCESS_CONDITIONS,
+        failure_conditions: Iterable[str] = DEFAULT_FAILURE_CONDITIONS,
+    ) -> AllNamespacedResource:
+        ...
+
+    def wait(
+        self,
+        res,
+        name: str,
+        *,
+        namespace=None,
+        success_conditions: Iterable[str] = DEFAULT_SUCCESS_CONDITIONS,
+        failure_conditions: Iterable[str] = DEFAULT_FAILURE_CONDITIONS,
+    ):
+        """Waits for specified conditions."""
+
+        kind = r.api_info(res).plural
+
+        success_conditions = list(success_conditions)
+        failure_conditions = list(failure_conditions)
+
+        for op, obj in self.watch(res, namespace=namespace):
+            if obj.metadata.name != name:
+                continue
+
+            if obj.status is None:
+                continue
+
+            if op == "DELETED":
+                raise Exception(f"{kind}/{name} was deleted while waiting for it")
+
+            try:
+                status = obj.status.to_dict()
+            except AttributeError:
+                status = obj.status
+
+            conditions = list(sorted(status['conditions'], key=lambda c: c['lastTransitionTime']))
+
+            # There may be multiple conditions with the same timestamp, so grab all with the most
```

Comment:
I think we should look at all conditions:
1. Maybe the condition did happen already before we exec the watch
2. Conditions can be modified by different components running on different nodes with different internal clock (for example kubelet and an operator). As a result picking the last `lastTransitionTime` may not always be accurate.